app/jobs/course/assessment/answer/auto_grading_job.rb

```diff
@@ -6,12 +6,15 @@ class Course::Assessment::Answer::AutoGradingJob < ApplicationJob
 
   # Performs the auto grading.
   #
+  # @param [String] redirect_to The path to be redirected after auto grading job was finished.
   # @param [Course::Assessment::Answer::AutoGrading] auto_grading The object to store the grading
   #   results into.
-  def perform_tracked(auto_grading)
+  def perform_tracked(auto_grading, redirect_to = nil)
```

Comment:
Yep, the main reason is that an extra SQL query is needed if we set the url outside.